### PR TITLE
correction cilium policy format

### PR DESCRIPTION
**Pull request by martijnvdp**
#### What this PR does / why we need it:
Fixes wrong formatting cilium policys
And updated the policys to a working and tested setup
changed the dd_url variable to the datadog.site variable 
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [ X] Chart Version bumped
- [ X] `CHANGELOG.md` has beed updated
- [ ] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/templates/agent-cilium-network-policy.yaml [hunk 1]
```diff
@@ -11,6 +11,24 @@ metadata:
...
+        app: {{ template "datadog.fullname" . }}
+        {{- if .Values.agents.podLabels }}
+        {{ toYaml .Values.agents.podLabels | indent 8 }}
+        {{- end }}
+    egress:
+      - toEntities:
+          - world
+        toPorts:
+          - ports:
+              - port: "51678"
```
Comment (clamoriniere) [inline charts/datadog/templates/agent-cilium-network-policy.yaml:26]: could you explain for which service the `51678` is used?

File: charts/datadog/templates/agent-cilium-network-policy.yaml [hunk 2]
```diff
@@ -50,12 +68,16 @@ specs:
         {{- end }}
     egress:
       - toFQDNs:
-          {{- if $.Values.datadog.dd_url}}
-          - matchName: {{ trimPrefix "https://" $.Values.datadog.dd_url }}
```
Comment (L3n41c) [inline charts/datadog/templates/agent-cilium-network-policy.yaml:54]: The `datadog.dd_url` parameter still exists: https://github.com/DataDog/helm-charts/blob/9611b4af7d929b7eb8792fd3e8ee347cb5bacc8c/charts/datadog/values.yaml#L67
We should support both `datadog.dd_url` and `datadog.site`.

File: charts/datadog/templates/agent-clusterchecks-cilium-network-policy.yaml [hunk 3]
```diff
@@ -41,8 +41,8 @@ specs:
         app: {{ template "datadog.fullname" . }}-clusterchecks
     egress:
       - toFQDNs:
-          {{- if $.Values.datadog.dd_url}}
-          - matchName: {{ trimPrefix "https://" $.Values.datadog.dd_url }}
```
Comment (L3n41c) [inline charts/datadog/templates/agent-clusterchecks-cilium-network-policy.yaml:45]: Here also, we should support both `datadog.dd_url` and `datadog.site`.

File: charts/datadog/templates/cluster-agent-cilium-network-policy.yaml [hunk 4]
```diff
@@ -50,10 +50,12 @@ specs:
         {{- end }}
     egress:
       - toFQDNs:
-          {{- if $.Values.datadog.dd_url}}
-          - matchName: {{ trimPrefix "https://" $.Values.datadog.dd_url }}
```
Comment (L3n41c) [inline charts/datadog/templates/cluster-agent-cilium-network-policy.yaml:54]: Here also we should support both `datadog_dd_url` and `datadog.site`.

File: charts/datadog/templates/agent-cilium-network-policy.yaml (hunk 1 above)
Comment (L3n41c) [inline charts/datadog/templates/agent-cilium-network-policy.yaml:26]: What is this port about ?

File: charts/datadog/templates/agent-cilium-network-policy.yaml [hunk 5]
```diff
@@ -50,12 +68,16 @@ specs:
...
+          - matchPattern: "*-app.agent.{{ $.Values.datadog.site }}"
+          - matchName: "agent-intake.logs.{{ $.Values.datadog.site }}"
+          - matchName: "process.{{ $.Values.datadog.site }}"
+          - matchName: "orchestrator.{{ $.Values.datadog.site }}"
           {{- else}}
           - matchPattern: "*-app.agent.datadoghq.com"
           - matchName: "agent-intake.logs.datadoghq.com"
           - matchName: "process.datadoghq.com"
+          - matchName: "orchestrator.datadoghq.com"
           {{- end}}
```
Comment (L3n41c) [inline charts/datadog/templates/agent-cilium-network-policy.yaml:81]:
```suggestion
          {{- if $.Values.datadog.dd_url}}
          - matchName: {{ trimPrefix "https://" $.Values.datadog.dd_url }}
          {{- end}}
          {{- if $.Values.datadog.site}}
          - matchPattern: "*-app.agent.{{ $.Values.datadog.site }}"
          - matchName: "agent-intake.logs.{{ $.Values.datadog.site }}"
          - matchName: "process.{{ $.Values.datadog.site }}"
          - matchName: "orchestrator.{{ $.Values.datadog.site }}"
          {{- else}}
          - matchPattern: "*-app.agent.datadoghq.com"
          - matchName: "agent-intake.logs.datadoghq.com"
          - matchName: "process.datadoghq.com"
          - matchName: "orchestrator.datadoghq.com"
          {{- end}}
```

**Review by L3n41c — COMMENTED** [review]
Thanks you very much for this contribution @martijnvdp !

I left few comments below.
I saw that the support of `datadog.dd_url` has been replaced by the support of `datadog.site`. It makes sense as I suppose that more users need `site` than `dd_url`. However, it would be nice to keep both.

You will also need to bump the version in the [`Chart.yaml`](https://github.com/DataDog/helm-charts/blob/master/charts/datadog/Chart.yaml#L3) and to add an entry in the [`CHANGELOG.md`](https://github.com/DataDog/helm-charts/blob/master/charts/datadog/CHANGELOG.md)

File: charts/datadog/templates/agent-cilium-network-policy.yaml (hunk 1 above)
Comment (martijnvdp) [reply charts/datadog/templates/agent-cilium-network-policy.yaml:26]: port of the ecs agent

Comment (martijnvdp) [reply charts/datadog/templates/agent-cilium-network-policy.yaml:26]: port of the ecs agent

**Comment by martijnvdp** [comment]
Thanks for the review i made the requested changes, let me know if this is ok

File: charts/datadog/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,8 @@
 # Datadog changelog
 
+## 2.15.2
+* Fix Cilium network policys.
```
Comment (L3n41c) [inline charts/datadog/CHANGELOG.md:4]:
```suggestion
* Fix Cilium network policies.
```

**Review by L3n41c — APPROVED** [review]

File: charts/datadog/CHANGELOG.md (hunk 6 above)
Comment (martijnvdp) [reply charts/datadog/CHANGELOG.md]: Thanks have changed it and upped the versions again

**Pull request merged**